yb-voyager/cmd/common.go

```diff
@@ -1176,20 +1176,36 @@ func getMigrationComplexityForOracle(schemaDirectory string) (string, error) {
 
 // TODO: consider merging all unsupported field with single AssessmentReport struct member as AssessmentIssue
 type AssessmentReport struct {
-	VoyagerVersion             string                                `json:"VoyagerVersion"`
-	TargetDBVersion            *ybversion.YBVersion                  `json:"TargetDBVersion"`
-	MigrationComplexity        string                                `json:"MigrationComplexity"`
-	SchemaSummary              utils.SchemaSummary                   `json:"SchemaSummary"`
-	Sizing                     *migassessment.SizingAssessmentReport `json:"Sizing"`
-	UnsupportedDataTypes       []utils.TableColumnsDataTypes         `json:"UnsupportedDataTypes"`
-	UnsupportedDataTypesDesc   string                                `json:"UnsupportedDataTypesDesc"`
-	UnsupportedFeatures        []UnsupportedFeature                  `json:"UnsupportedFeatures"`
-	UnsupportedFeaturesDesc    string                                `json:"UnsupportedFeaturesDesc"`
-	UnsupportedQueryConstructs []utils.UnsupportedQueryConstruct     `json:"UnsupportedQueryConstructs"`
-	UnsupportedPlPgSqlObjects  []UnsupportedFeature                  `json:"UnsupportedPlPgSqlObjects"`
-	MigrationCaveats           []UnsupportedFeature                  `json:"MigrationCaveats"`
-	TableIndexStats            *[]migassessment.TableIndexStats      `json:"TableIndexStats"`
-	Notes                      []string                              `json:"Notes"`
+	VoyagerVersion      string                                `json:"VoyagerVersion"`
+	TargetDBVersion     *ybversion.YBVersion                  `json:"TargetDBVersion"`
+	MigrationComplexity string                                `json:"MigrationComplexity"`
+	SchemaSummary       utils.SchemaSummary                   `json:"SchemaSummary"`
+	Sizing              *migassessment.SizingAssessmentReport `json:"Sizing"`
+	Issues              []AssessmentIssue                     `json:"-"` // disabled in reports till corresponding UI changes are done(json and html reports)
+	TableIndexStats     *[]migassessment.TableIndexStats      `json:"TableIndexStats"`
+	Notes               []string                              `json:"Notes"`
+
+	// fields going to be deprecated
+	UnsupportedDataTypes       []utils.TableColumnsDataTypes     `json:"UnsupportedDataTypes"`
+	UnsupportedDataTypesDesc   string                            `json:"UnsupportedDataTypesDesc"`
+	UnsupportedFeatures        []UnsupportedFeature              `json:"UnsupportedFeatures"`
+	UnsupportedFeaturesDesc    string                            `json:"UnsupportedFeaturesDesc"`
+	UnsupportedQueryConstructs []utils.UnsupportedQueryConstruct `json:"UnsupportedQueryConstructs"`
+	UnsupportedPlPgSqlObjects  []UnsupportedFeature              `json:"UnsupportedPlPgSqlObjects"`
+	MigrationCaveats           []UnsupportedFeature              `json:"MigrationCaveats"`
+}
+
+type AssessmentIssue struct {
```

Comment:
@sanyamsinghal can you add comments to each field, giving an example of what they might look like?

Reply:
Category - `Feature`, `Migration Caveat`. `Query Constructs`, `Datatype`
TypeName - `Gin Indexes`, `Regex Functions`, ...
TypeDescription - TODO: add if available or add for required ones
Impact: Last meeting we discuss it as Level-1, Leve-2, Leve-3... But i am thinking we can have it as `Minor` `Moderate` `Significant` also

ObjectType - it can be empty in some cases, for categories like unsupported datatypes thinking of the object type as datatype for eg: `geometry`
Basically thinking of the datatype as object type

We can discuss this more in next PR, for now i just defined a struct. @makalaaneesh

Reply:
got it, yeah just wanted you to add comments to code next to the field 👍

Reply:
sure, will add.

Reply:
@makalaaneesh let me know what do you think about 
> ObjectType - it can be empty in some cases, for categories like unsupported datatypes thinking of the object type as datatype for eg: geometry
Basically thinking of the datatype as object type